### ISSUE TITLE
[Feature] enable basic nginx monitoring capabilities

### ISSUE THREAD

**Issue by voelzmo** [cloudfoundry/bosh#2058]
**Is your feature request related to a problem? Please describe.**
Figuring out when to scale nginx is terrible right now. We're waiting for user errors to happen and then look at options for scaling.

**Describe the solution you'd like**
Reading nginx metrics such that we can more easily see if we need to scale e.g. nginx workers. One option is to enable the `ngx_http_stub_status` module, which provides simple metrics, such as number of active connections, number of accepted vs. handles requests, etc. http://nginx.org/en/docs/http/ngx_http_stub_status_module.html

It exposes a web page listing information like this
```
Active connections: 291 
server accepts handled requests
 16630948 16630948 31070465 
Reading: 6 Writing: 179 Waiting: 106 
```

Let's configure nginx with `--with-http_stub_status_module` when compiling it and provide an option in the manifest which adds an additional site configuration like this
```
server {
    location /stats {
        stub_status;
        access_log off;
        allow 127.0.0.1;
        deny all;
    }
}
```

If you want to forward data to a different source, you can co-locate a binary. Parsers/forwarders exist for a number of monitoring tools,
* collectD https://collectd.org/wiki/index.php/Plugin:nginx
* metricbeat https://www.elastic.co/guide/en/beats/metricbeat/current/metricbeat-metricset-nginx-stubstatus.html
* influxdb telegraf https://github.com/influxdata/telegraf/tree/master/plugins/inputs/nginx

**Describe alternatives you've considered**
Other means of monitoring seems inefficient.

**Comment by dpb587-pivotal** [comment]
This makes sense to me; it's a popular endpoint to expose, regardless of what telemetry monitoring/forwarding you end up using. For what it's worth, I think it should be exposed on a localhost-only bind.

**Comment by voelzmo** [comment]
> For what it's worth, I think it should be exposed on a localhost-only bind.

I think so as well, that's why I included the sample configuration above, which should achieve this
```
server {
    location /stats {
        stub_status;
        access_log off;
        allow 127.0.0.1;
        deny all;
    }
}
```

If we are agreeing this is something useful to do and have, I'd put this into our backlog and prioritize it.

**Comment by dpb587-pivotal** [comment]
To recap an offline conversation from this group, it sounds like this would be a good change to make. It will help as we add expose more of this sort of metrics through the product(s).

**Comment by dpb587-pivotal** [comment]
Closing - available since [v268.3.0](https://github.com/cloudfoundry/bosh/releases/tag/v268.3.0).
